solid/src/main/java/com/inrupt/client/solid/SolidClient.java

```diff
@@ -280,14 +285,15 @@ public <T extends Resource> CompletionStage<Void> delete(final T resource, final
         defaultHeaders.firstValue(USER_AGENT).ifPresent(agent -> builder.setHeader(USER_AGENT, agent));
         headers.firstValue(USER_AGENT).ifPresent(agent -> builder.setHeader(USER_AGENT, agent));
 
-        return client.send(builder.build(), Response.BodyHandlers.ofByteArray()).thenApply(res -> {
-            if (isSuccess(res.statusCode())) {
-                return null;
-            } else {
-                throw SolidClientException.handle("Unable to delete resource", resource.getIdentifier(),
-                        res.statusCode(), res.headers(), new String(res.body(), UTF_8));
-            }
-        });
+        return client.send(
+            builder.build(),
+            Response.BodyHandlers.throwOnError(
+                Response.BodyHandlers.ofByteArray(),
+                (r) -> isSuccess(r.statusCode()),
+                httpExceptionMapper("Deleting resource " + resource.getIdentifier() + " failed.")
```

Comment:
Potential PII in message. Should the message be left at the previous `Unable to delete resource`. The resource URI will be `SolidClientException`

Reply:
Removed with https://github.com/inrupt/solid-client-java/pull/1157/commits/c03c1d93fd1dfec6644150ecf299f5c027472e59